browser/brave_content_browser_client.cc

```diff
@@ -198,6 +213,39 @@ void BindCosmeticFiltersResources(
       std::move(receiver));
 }
 
+#if BUILDFLAG(BRAVE_ADS_ENABLED)
+void BindBraveAdsDriver(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<brave_ads::mojom::BraveAdsDriver> receiver) {
+  auto* context = frame_host->GetBrowserContext();
+  auto* profile = Profile::FromBrowserContext(context);
+  if (brave::IsRegularProfile(profile)) {
```

Comment:
you should not be making this check here. The factories for ads and rewards return nullptr for contexts they are not valid for.

Reply:
moved factories code to BraveAdsDriver

Reply:
this is still wrong and `BraveAdsHostPrivate` isn't needed. This check handles it https://github.com/brave/brave-core/pull/9611/files#diff-03b16e27b577f555026dfd838511fd4ee0d1a2aaede7f59cb49e7126a9cdd901R51